api/envoy/extensions/filters/network/zookeeper_proxy/v3/zookeeper_proxy.proto

```diff
@@ -39,4 +39,50 @@ message ZooKeeperProxy {
   //
   // if that is set. If it isn't, ZooKeeper's default is also 1Mb.
   google.protobuf.UInt32Value max_packet_bytes = 3;
+
+  // List of latency thresholds for opcodes to decide the fast/slow responses and emit metrics (used for error budget calculation).
```

Comment:
Can multiple entries have the same Opcode?
If not, please clarify the comment, and add a test that validates that the config is rejected.

Reply:
I've added a valid unit test with duplicated opcode entries.

Reply:
FWIW, PGV (protoc-gen-validate) cannot check this, but the code can, and decide to reject (throw an exception) in this case. As an example: [route validity](https://github.com/envoyproxy/envoy/blob/18f89086ef5341b7d152eb3f1f5b16e2ea3749eb/source/common/router/config_impl.cc#L156-L160) is one of the cases where a config is rejected, and cannot be validated by PGV.

Overriding is a valid approach, with the downside of not detecting configuration smells until it becomes late.

Whatever approach you choose, please update the comments to describe this.

Reply:
Thanks, I prefer check the duplicated opcode in the code. Will make the code/test/comment changes accordingly.